web/pages/notifications.tsx

```diff
@@ -237,65 +265,52 @@ function NotificationGroupItem(props: {
   return (
     <NotificationGroupItemComponent
       notifications={notifications}
-      highlighted={highlighted}
-      setHighlighted={setHighlighted}
       header={header}
     />
   )
 }
 
 export function NotificationGroupItemComponent(props: {
   notifications: Notification[]
-  highlighted: boolean
-  setHighlighted: (highlighted: boolean) => void
   header: ReactNode
   isIncomeNotification?: boolean
   className?: string
 }) {
-  const {
+  const { notifications, className, header, isIncomeNotification } = props
+  const [readNotifications, unreadNotifications] = partition(
     notifications,
-    className,
-    setHighlighted,
-    header,
-    isIncomeNotification,
-  } = props
-
-  const needsExpanding = notifications.length > NUM_SUMMARY_LINES
-  const [expanded, setExpanded] = useState(
-    notifications.length <= NUM_SUMMARY_LINES
+    (n: Notification) => n.isSeen
   )
+  const numRead = readNotifications.length
+  const numUnread = unreadNotifications.length
+  const numReadShown = NUM_SUMMARY_LINES - numUnread
+
+  const needsExpanding = numRead > 0 && notifications.length > NUM_SUMMARY_LINES
+  const [expanded, setExpanded] = useState(false)
   const onExpandHandler = (event: React.MouseEvent<HTMLDivElement>) => {
     if (event.ctrlKey || event.metaKey) return
     setExpanded(!expanded)
   }
 
-  useEffect(() => {
-    if (expanded) setHighlighted(false)
-  }, [expanded, setHighlighted])
-
+  const shownArray = expanded
```

Comment:
> grouped notifications now previews all unread notifications, even if unread notifications exceeds 3 notifications

I don't think it's a good idea to show more than 3 unread notifications on a market at a time. Otherwise markets with a lot of activity will take up the whole page.